icu4c/source/data/unidata/delete_build_headers.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# © 2021 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+# This script deletes newly created header files produced by create_build_headers.sh ,
```

Comment:
This is not quite what I imagined.

First, I assumed that we would be accumulating a list of generated files. "bazelisk clean" is much better!

Second, I suggested deleting the to-be-generated files before "bazelisk build" to ensure that the build process cannot rely on the to-be-generated files while generating. Therefore, we don't want to "git restore" them either. "git restore" seems only useful when something went wrong and we want to get back to a source tree that works with the "normal" build system.

I suggest you start a TODO section near the top of unidata/changes.txt and put the "bazelisk clean" line there with a one-liner about deleting the to-be-generated files. We can clean that up later.

As for "git restore", can you just `git restore .` (or something) at top level and ? If so, we don't need this .sh file here at all.

Or do we need a script with a hardcoded list? In that case, I would rename this .sh to restore_output_files.sh or simply restore.sh.

Reply:
Reviewed together; I had still misunderstood. "bazelisk clean" deletes the *output* files, not the files in the source tree. We want to delete the latter, to guarantee that the bazel build does not pick up old versions of files that it generates.

Agreed on
- rename to clean.sh
- no "bazelisk clean" here but might mention it in unidata/changes.txt
- change "git restore" to "rm"